packages/indexer-common/src/allocations/types.ts

```diff
@@ -26,3 +34,143 @@ export enum AllocationStatus {
   FINALIZED = 'Finalized',
   CLAIMED = 'Claimed',
 }
+
+export interface MonitorEligibleAllocationsOptions {
+  indexer: Address
+  logger: Logger
+  networkSubgraph: NetworkSubgraph
+  interval: number
+  protocolNetwork: string
+}
+
+export const monitorEligibleAllocations = ({
+  indexer,
+  logger: parentLogger,
+  networkSubgraph,
+  protocolNetwork,
+  interval,
+}: MonitorEligibleAllocationsOptions): Eventual<Allocation[]> => {
+  const logger = parentLogger.child({ component: 'AllocationMonitor' })
+
+  const refreshAllocations = async (
+    currentAllocations: Allocation[],
+  ): Promise<Allocation[]> => {
+    logger.debug('Refresh eligible allocations', {
+      protocolNetwork,
+    })
+
+    try {
+      const currentEpochResult = await networkSubgraph.query(gql`
+        query {
+          graphNetwork(id: "1") {
```

Comment:
This doesn't seem to be a "type"

Reply:
would you name the file utils? seems more like it for allocations

Reply:
"eventuals" or "monitor"